### DISCUSSION TITLE
z2mqtt not stable at all

### DISCUSSION THREAD

**Discussion by sygys** [Koenkk/zigbee2mqtt#14206] [General]
Hey Koen,

Its fun to have seen you on stage at the tweakers meet up. i didnt know you lived in the same city that i do.

The reason im posting this is because i have been running into allot of issues the last (few) years. for the most part with z2mqtt. I find it to be very unstable and i cant really tell what is going on. So i would like to start a discussion to see if we can fix the unstable state z2m is in at the moment instead of adding more devices everytime.

For the most part i guess z2m could be stable but as soon as something changes in the network everything goes to #%$@. For example last night i had a power outage for a short time. now a  few blitzwolf smartplugs all of a sudden dont report any states anymore. How is that possible? the error is something like this:

`Publish 'get' 'state' to 'Mancave NUC10 smartplug' failed: 'Error: Read 0x04cd15fffe02cec2/1 genOnOff(["onOff"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205))'`

Also when restarting z2m strange things happen. its almost like the whole routing of my zigbee network needs to be done all over because lights are slow to respond sometimes. Temperature sensors are falling off the network when home assistant is down for a few minutes. hue lights falling off the network for no reason becomming fully unresponsive and needing a hard reset. 

I have a network of nearly 130 devices (over 100 routers) and using a tube zigbee router with CC2652P2 chipset. i installed the latest firmware on it a few weeks back.

Are these problems caused because of the big diversity of devices that cannot work together good enough?

Im also seeing that when i have routers in z2mqtt that are offline because they dont have power the network becomes unstable. its almost as if z2m keeps trying to let routes go through that device while its not possible.  the 'No network route' (205))' error that i posted above seems to be the proof... z2m seems to not make new connections when devices go offline. If this is the case then with large networks the stability will begin to drop because there is always a device that misses a communication now and then due to interference or so.

It would be great if we could talk some more and try to solve the problems.

Dont get me wrong i understand that troubleshooting can be hard and every setup is different. but im seeing a structural problem here.

I guess the biggest issues start when a router cant be reached because it has no power. then the whole network becomes unstable to the point where things seems to drop out or react late. (sometimes 10 seconds after motion is detected a light goes on).

**Comment by sygys** [comment]
The whole reason this is comming to my attention is a almost fire in our fusebox. Because of that i had to disable 1 fase out of the 3 allot of zigbee devices around the house dont have power at the moment. 

What im seeing now with more testing, is that routes dont seem to get redirected or in a very slow matter. So for some reason z2m keeps making routes to devices that are allready offline for several days now and it sometimes takes over 10 seconds before a message reaches an online device. Shouldnt z2m with the coordinator make new routes and stick to them as long as these device stay offline? 

I personally think z2m needs allot of work to get all this stable. Yes it works when everything stays online and you have absolutely no interference but that is almost never the case. as soon as devices go offline the whole network is struggles and for some reason even after a few days its not getting back on its feet.

**Reply by Koenkk** [reply]
I've recently worked with someone with a larger network (+- 150 devices) to improve the firmware. In his situation we managed to greatly improve performance. Can you check if this is also the case for you with this fw: https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin ? For the Tube coordinator you will need `CC1352P2_CC2652P_launchpad_coordinator_20220219.zip`.

If it doesn't improve we can debug further.

**Reply by jerrm** [reply]
Would it be fair to say most of this looks more like firmware/mesh/device issues, and not really a z2m issue? 

The extreme example would be older Aqara door sensors.  If the router they attach to goes down, they will take hours (8+ in my testing) to establish an alternate connection.  Not a damned thing z2m can do about that.  

I would assume it is the firmware generating the 'No network route' error and z2m is only reporting what it receives from the coordinator?

Does z2m really have much (any?) control over the layer 3(ish) and below stuff?

**Reply by Koenkk** [reply]
It differs per case but mostly it is the mesh or the firmware.

**Reply by sygys** [reply]
It sounds very good to improve the network with firmware. I'll try the new firmware soon. The funny thing seems to be that my current firmware does work pretty good with everything online. But as soon as I disconnect a router things seems to break.

**Reply by sygys** [reply]
Been a while to report back. But today i looked into this and i allready have this version of firmware installed on the coordinator. 

Today i shortly cut power to a light bulb on accident (hue RGB bulb) and now it has fallen off the network as well. the only way to get it back on the network is to use a dimmer switch and holding the on and off buttons for 10 seconds to do the most agressive reset existing to being able to pair the bulb to the network again. I also have a few innr spots that occasionally stop working until i cut the power to them.

A few days back i had the issue where a hue filament bulb did respond to group commands setup with z2mqtt but wont react to any individual commands giving a timeout error in the logs of 10000 ms

I dont know whats wrong here but my network is very unstable. i have around i guess 80 routers in a normal house. in every room i have atleast 2 and mostly 3 or more routers. So a problem with range or unstabel routes shouldnt occur except if z2m isnt making them.

I also monitor the z2m connection so that i get a message when the whole network goes down. This occurs not often but every now and then all devices are offline for a second and come back on.

Im building a smart home for over 3 years now. and as long as i have been using z2m im having issues. they do come and go with z2m updates so it needs to be something with z2m not routing everything like it should. But i have not a clue how to troubleshoot this.

At the moment i see no other way then to remove all zigbee devices one by one replacing them with zwave which is a 1000 times more stable.

**Comment by jerrm** [comment]
I still doubt the problems are actually z2m issues.  The mesh is mostly managed auto-magically at the coordinator and below level. I don't think Z2M itself really picks or manages mesh routes to devices.

First make sure you've ruled out, or at least minimized as much as possible, any RF interference issues (wifi/zigbee channel selections, usb3 issues, antenna placement, etc).

If still unsuccessful, I would set up a second zigbee net and move devices over slowly and methodically to see if there is a specific point or device where the new net breaks or the old net starts working as expected.  

$30-$150 for new coordinators to test with is worth the cost compared to dumping thousands of dollars in other equipment.  Heck, before dumping everything I would be trying a new compatible CC2652 coordinator on the existing z2m instance, test building out second z2m instance with another cc2652, and maybe even test a third z2m instance with an ezsp stick.  If running home assistant I'd give ZHA a try with both sticks as well.

**Comment by notabenex** [comment]
I have actually 118 devices with ziigbee2mqtt (Legrand, HUE, Aqara, Xiaomi....) working abslutely flawlessly!

**Comment by sygys** [comment]
Also when I force remove a crashed light and not reboot z2m and then repair it. It will be added to z2m again but with the same errors as before. So i have to remove the light reboot z2m and then repair it else it will never work

**Comment by sygys** [comment]
Well flawless is something that everyone experience different. When a light 4esponds a second late then for me the network isn't flawless. Also when a Xiaomi door sensor misses it's state in z2m then the network is in my opinion broken and unreliable. This also doesn't explain why a light responds to group commands in z2m but doesn't react to individual commands. This is clearly something to blame z2m.

Also z2m manages all routes. The coordinator incase of z2m only makes the communication

**Reply by grimelke001** [reply]
Installed Raspian (Pi 4, 4Gb, 128Gb boot SSD) with HA,  some 20 routers, 30 devices total (Aqara, sonoff, Aubess and some no-brand), USB Dongle Sonoff efr32mg21, first tryed ZHA it was flawless but not all atrib from the devices i could use.
So a few days ago i dropped ZHA and installed  z2mqtt.
Sygys formulate the problems, my experience is the same.
I have the same issues from day one and ill hop tomorrow morning back to ZHA while this is so disturbing if you dont have a relayable network!  a second later respons is not a problem!

btw ZHA works on channel 15 and Z2mqtt on channel 11 with latest firmware.

**Comment by sygys** [comment]
At the moment I'm replacing all essential devices with wifi ones from Shelly. And dropping the Zigbee ones z2mqtt is far from stable. I'm also planning to revert back to the hue bridge eventually which has always been rock solid. The problem i guess is the management of the network. As soon as there is the slightest Interference the whole network is a mess. Because the rerouting is poor. Also it chooses very poor routes from one side of the house to the other while 20 routers are in between. So the network isn't choosing the best routes. Also when a router disconnects the routes it managed aren't rerouted or takes several hours/ days.

**Comment by notabenex** [comment]
Currently I have 130 zigbee devices. Including legrand, hue, aqara..
Everything smooth with rpi4 and HA

Dne čt 24. 11. 2022 19:31 uživatel sygys ***@***.***> napsal:
